src/osx/carbon/utilscocoa.mm

```diff
@@ -613,6 +613,27 @@ void  wxMacCocoaShowCursor()
 {
     [NSCursor unhide];
 }
+
+WX_NSCursor wxMacCocoaCreateCursorFromResource(const wxString &cursor_file, wxBitmapType flags)
+{
+    NSString *path;
+    NSBundle *bundle = [NSBundle mainBundle];
+    if( flags == wxBITMAP_TYPE_MACCURSOR_RESOURCE )
+        path = [bundle pathForResource:wxCFStringRef( cursor_file ).AsNSString() ofType:@"cur"];
+    if( path )
+    {
+        NSURL *url = [NSURL fileURLWithPath:path];
+        CGImageSourceRef image = CGImageSourceCreateWithURL( (__bridge CFURLRef) url, nil );
+        NSDictionary *properties = (__bridge NSDictionary *)CGImageSourceCopyPropertiesAtIndex( image, 0, nil );
+        CGFloat x = [[properties objectForKey:@"hotspotX"]floatValue];
```

Comment:
I think you could use a `wxCFDictionaryRef` here, no need to go to `NSDictionary`. And then a GetValue("hotspotX").GetValue(&x)

Reply:
@csomor ,
I tried following code:

```
        wxCFDictionaryRef properties( (wxCFDictionaryRef) CGImageSourceCopyPropertiesAtIndex( image, 0, nil ) );
        properties.GetValue( "hotspotX" ).GetValue( &x );
        properties.GetValue( "hotspotY" ).GetValue( &y );

and it failed on the retrieval of the x - BAD-EXEC exception.

Any suggestions?

```

Reply:
A PNG does not have a hotspot, that's why you have to indicate it in the other parameters of the wxCursor from file parameter, an CUR does however, so the wxImage constructed from it already does know about the hotspot. We have everything in the our code already.